// Code generated by protoc-gen-validate. DO NOT EDIT.
// source: api/stock/v1/callback.proto

package v1

import (
	"bytes"
	"errors"
	"fmt"
	"net"
	"net/mail"
	"net/url"
	"regexp"
	"sort"
	"strings"
	"time"
	"unicode/utf8"

	"google.golang.org/protobuf/types/known/anypb"
)

// ensure the imports are used
var (
	_ = bytes.MinRead
	_ = errors.New("")
	_ = fmt.Print
	_ = utf8.UTFMax
	_ = (*regexp.Regexp)(nil)
	_ = (*strings.Reader)(nil)
	_ = net.IPv4len
	_ = time.Duration(0)
	_ = (*url.URL)(nil)
	_ = (*mail.Address)(nil)
	_ = anypb.Any{}
	_ = sort.Sort
)

// Validate checks the field values on QueryPreparedRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *QueryPreparedRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on QueryPreparedRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// QueryPreparedRequestMultiError, or nil if none found.
func (m *QueryPreparedRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *QueryPreparedRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return QueryPreparedRequestMultiError(errors)
	}

	return nil
}

// QueryPreparedRequestMultiError is an error wrapping multiple validation
// errors returned by QueryPreparedRequest.ValidateAll() if the designated
// constraints aren't met.
type QueryPreparedRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m QueryPreparedRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m QueryPreparedRequestMultiError) AllErrors() []error { return m }

// QueryPreparedRequestValidationError is the validation error returned by
// QueryPreparedRequest.Validate if the designated constraints aren't met.
type QueryPreparedRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e QueryPreparedRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e QueryPreparedRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e QueryPreparedRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e QueryPreparedRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e QueryPreparedRequestValidationError) ErrorName() string {
	return "QueryPreparedRequestValidationError"
}

// Error satisfies the builtin error interface
func (e QueryPreparedRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sQueryPreparedRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = QueryPreparedRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = QueryPreparedRequestValidationError{}

// Validate checks the field values on QueryPreparedReply with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *QueryPreparedReply) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on QueryPreparedReply with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// QueryPreparedReplyMultiError, or nil if none found.
func (m *QueryPreparedReply) ValidateAll() error {
	return m.validate(true)
}

func (m *QueryPreparedReply) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return QueryPreparedReplyMultiError(errors)
	}

	return nil
}

// QueryPreparedReplyMultiError is an error wrapping multiple validation errors
// returned by QueryPreparedReply.ValidateAll() if the designated constraints
// aren't met.
type QueryPreparedReplyMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m QueryPreparedReplyMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m QueryPreparedReplyMultiError) AllErrors() []error { return m }

// QueryPreparedReplyValidationError is the validation error returned by
// QueryPreparedReply.Validate if the designated constraints aren't met.
type QueryPreparedReplyValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e QueryPreparedReplyValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e QueryPreparedReplyValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e QueryPreparedReplyValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e QueryPreparedReplyValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e QueryPreparedReplyValidationError) ErrorName() string {
	return "QueryPreparedReplyValidationError"
}

// Error satisfies the builtin error interface
func (e QueryPreparedReplyValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sQueryPreparedReply.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = QueryPreparedReplyValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = QueryPreparedReplyValidationError{}

// Validate checks the field values on DeleteCacheRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the first error encountered is returned, or nil if there are no violations.
func (m *DeleteCacheRequest) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on DeleteCacheRequest with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// DeleteCacheRequestMultiError, or nil if none found.
func (m *DeleteCacheRequest) ValidateAll() error {
	return m.validate(true)
}

func (m *DeleteCacheRequest) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if utf8.RuneCountInString(m.GetKey()) < 1 {
		err := DeleteCacheRequestValidationError{
			field:  "Key",
			reason: "value length must be at least 1 runes",
		}
		if !all {
			return err
		}
		errors = append(errors, err)
	}

	if len(errors) > 0 {
		return DeleteCacheRequestMultiError(errors)
	}

	return nil
}

// DeleteCacheRequestMultiError is an error wrapping multiple validation errors
// returned by DeleteCacheRequest.ValidateAll() if the designated constraints
// aren't met.
type DeleteCacheRequestMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m DeleteCacheRequestMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m DeleteCacheRequestMultiError) AllErrors() []error { return m }

// DeleteCacheRequestValidationError is the validation error returned by
// DeleteCacheRequest.Validate if the designated constraints aren't met.
type DeleteCacheRequestValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DeleteCacheRequestValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DeleteCacheRequestValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DeleteCacheRequestValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DeleteCacheRequestValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DeleteCacheRequestValidationError) ErrorName() string {
	return "DeleteCacheRequestValidationError"
}

// Error satisfies the builtin error interface
func (e DeleteCacheRequestValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDeleteCacheRequest.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DeleteCacheRequestValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DeleteCacheRequestValidationError{}

// Validate checks the field values on DeleteCacheReply with the rules defined
// in the proto definition for this message. If any rules are violated, the
// first error encountered is returned, or nil if there are no violations.
func (m *DeleteCacheReply) Validate() error {
	return m.validate(false)
}

// ValidateAll checks the field values on DeleteCacheReply with the rules
// defined in the proto definition for this message. If any rules are
// violated, the result is a list of violation errors wrapped in
// DeleteCacheReplyMultiError, or nil if none found.
func (m *DeleteCacheReply) ValidateAll() error {
	return m.validate(true)
}

func (m *DeleteCacheReply) validate(all bool) error {
	if m == nil {
		return nil
	}

	var errors []error

	if len(errors) > 0 {
		return DeleteCacheReplyMultiError(errors)
	}

	return nil
}

// DeleteCacheReplyMultiError is an error wrapping multiple validation errors
// returned by DeleteCacheReply.ValidateAll() if the designated constraints
// aren't met.
type DeleteCacheReplyMultiError []error

// Error returns a concatenation of all the error messages it wraps.
func (m DeleteCacheReplyMultiError) Error() string {
	var msgs []string
	for _, err := range m {
		msgs = append(msgs, err.Error())
	}
	return strings.Join(msgs, "; ")
}

// AllErrors returns a list of validation violation errors.
func (m DeleteCacheReplyMultiError) AllErrors() []error { return m }

// DeleteCacheReplyValidationError is the validation error returned by
// DeleteCacheReply.Validate if the designated constraints aren't met.
type DeleteCacheReplyValidationError struct {
	field  string
	reason string
	cause  error
	key    bool
}

// Field function returns field value.
func (e DeleteCacheReplyValidationError) Field() string { return e.field }

// Reason function returns reason value.
func (e DeleteCacheReplyValidationError) Reason() string { return e.reason }

// Cause function returns cause value.
func (e DeleteCacheReplyValidationError) Cause() error { return e.cause }

// Key function returns key value.
func (e DeleteCacheReplyValidationError) Key() bool { return e.key }

// ErrorName returns error name.
func (e DeleteCacheReplyValidationError) ErrorName() string { return "DeleteCacheReplyValidationError" }

// Error satisfies the builtin error interface
func (e DeleteCacheReplyValidationError) Error() string {
	cause := ""
	if e.cause != nil {
		cause = fmt.Sprintf(" | caused by: %v", e.cause)
	}

	key := ""
	if e.key {
		key = "key for "
	}

	return fmt.Sprintf(
		"invalid %sDeleteCacheReply.%s: %s%s",
		key,
		e.field,
		e.reason,
		cause)
}

var _ error = DeleteCacheReplyValidationError{}

var _ interface {
	Field() string
	Reason() string
	Key() bool
	Cause() error
	ErrorName() string
} = DeleteCacheReplyValidationError{}
